rook/main.py

```diff
@@ -45,28 +47,90 @@
 
-if args.load_average > 0 and hasattr(os, "getloadavg"):
-  #Note that this is also done before starting each test
-  while os.getloadavg()[0] > args.load_average:
-    print("Load Average too high, waiting ", os.getloadavg()[0])
-    time.sleep(1.0)
+if socket.gethostname() == "VDI-RavenW7-01":
+  print("Overridding load average")
+  args.load_average = 4.0
 
-def load_average_adapter(function):
-  """
-    Adapts function to not start until load average is low enough
-    @ In, function, function, function to call
-    @ Out, new_func, function, function that checks load average before running
-  """
-  def new_func(data):
+if hasattr(os, "getloadavg"):
+  def load_average_adapter(function):
+    """
+      Adapts function to not start until load average is low enough
+      @ In, function, function, function to call
+      @ Out, new_func, function, function that checks load average before running
+    """
+    def new_func(data):
+      """
+        function that waits until load average is lower.
+        @ In, data, Any, data to pass to function
+        @ Out, result, result of running function on data
+      """
+      while os.getloadavg()[0] > args.load_average:
+        time.sleep(1.0)
+      return function(data)
+    return new_func
+
+else:
+  import psutil
+
+  class LoadClass(threading.Thread):
+    """
+      This class keeps track of the one second load average
     """
-      function that waits until load average is lower.
-      @ In, data, Any, data to pass to function
-      @ Out, result, result of running function on data
+
+    def __init__(self):
+      """
+        Initialize this class
+        @ In, None
+        @ Out, None
+      """
+      threading.Thread.__init__(self)
+      self.__load_avg = psutil.cpu_percent(1.0)*psutil.cpu_count()/100.0
+      self.__load_lock = threading.Lock()
+      self.daemon = True #Exit even if this thread is running.
+
+    def run(self):
+      """
+        Run forever, grabbing the load average every second
+        @ In, None
+        @ Out, None
+      """
+      while True:
+        load_avg = psutil.cpu_percent(1.0)*psutil.cpu_count()/100.0
+        with self.__load_lock:
+          self.__load_avg = load_avg
+
+    def get_load_average(self):
+      """
+        Get the most recent load average
+        @ In, None,
+        @ Out, float value for load average (average number of processors running)
+      """
+      load_avg = -1
+      with self.__load_lock:
+        load_avg = self.__load_avg
+      return load_avg
+
+  load = LoadClass()
+  load.start()
+
+  def load_average_adapter(function):
+    """
+      Adapts function to not start until load average is low enough
+      @ In, function, function, function to call
+      @ Out, new_func, function, function that checks load average before running
     """
-    while os.getloadavg()[0] > args.load_average:
-      time.sleep(1.0)
-    return function(data)
-  return new_func
+    def new_func(data):
```

Comment:
same as above

Reply:
This actually creates a ```new_func``` for every single test when using the load adaption.  This was the cleanest solution I found for doing the load average for every test since sometimes (such as when -l is not given or when using qsubs) nothing needs to be done.